featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -965,6 +995,7 @@ def _get_matching_inputs(
         primitive_options,
         require_direct_input=False,
         feature_filter=None,
+        transform_stacking=False,
```

Comment:
`transform_stacking` as parameter name is confusing.  Currnetly, if `transform_stacking` is `True`, features that stack on top of transform features are not allowed, and when `transform_stacking` is `False`, stacking on transform feature is allowed.

Also, sinces `check_transform_stacking` excludes any combination of inputs where any input is a transform feature or a direct feature of a transform feature, I think we can filter out those two groups of features as valid input types, period, before creating input combinations (either in `_features_by_type` or by making `check_transform_stacking` a `feature_filter` for `_get_matching_inputs` when called from `_build_transform_features`

Reply:
I'm going to take the feature_filter approach and see how that plays out